src/core/sagas/recommendations.ts

```diff
@@ -109,48 +111,61 @@ export namespace Tasks {
-          );
-          yield effects.put(<any>[
-            flux.actions.receivePastPurchaseAllRecordCount(results.totalRecordCount),
-            flux.actions.receivePastPurchaseRefinements(navigations),
-          ]);
-        } else {
-          yield effects.put(<any>[
-            flux.actions.updatePastPurchasePageSize(pageSize),
-            flux.actions.receivePastPurchasePage(
-              SearchAdapter.extractRecordCount(results.totalRecordCount),
-              PageAdapter.currentPage(request.skip, pageSize),
-              pageSize
-            ),
-            flux.actions.receivePastPurchaseCurrentRecordCount(results.totalRecordCount),
-            flux.actions.receivePastPurchaseProducts(SearchAdapter.augmentProducts(results)),
-            flux.actions.receivePastPurchaseTemplate(SearchAdapter.extractTemplate(results.template))
-          ]);
-          flux.replaceState(utils.Routes.PAST_PURCHASE);
-        }
+        yield effects.put(<any>[
+          flux.actions.updatePastPurchasePageSize(pageSize),
+          flux.actions.receivePastPurchasePage(
+            SearchAdapter.extractRecordCount(results.totalRecordCount),
+            PageAdapter.currentPage(request.skip, pageSize),
+            pageSize
+          ),
+          flux.actions.receivePastPurchaseCurrentRecordCount(results.totalRecordCount),
+          flux.actions.receivePastPurchaseProducts(SearchAdapter.augmentProducts(results)),
+          flux.actions.receivePastPurchaseTemplate(SearchAdapter.extractTemplate(results.template)),
+          flux.actions.receivePastPurchaseRefinements(
+            SearchAdapter.pruneRefinements(SearchAdapter.combineNavigations(results), state)
+          ),
+        ]);
+        flux.replaceState(utils.Routes.PAST_PURCHASE);
       }
     } catch (e) {
       return effects.put(flux.actions.receivePastPurchaseProducts(e));
     }
   }
 
+  export function* fetchPastPurchaseNavigations(flux: FluxCapacitor, action: Actions.FetchPastPurchaseNavigations) {
+    try {
+      const pastPurchaseSkus: Store.PastPurchases.PastPurchaseProduct[] = yield effects.select(Selectors.pastPurchases);
+      if (pastPurchaseSkus.length > 0) {
+        const state = yield effects.select();
+        const pastPurchasesFromSkus = Tasks.buildRequestFromSkus(flux, pastPurchaseSkus);
+        const request = pastPurchaseProductsRequest.composeRequest(state, {
+          ...pastPurchasesFromSkus,
+          ...action.payload.request
+        });
+        const results = yield effects.call(Requests.search, flux, request);
+        const navigations = PastPurchaseAdapter.pastPurchaseNavigations(
+          yield effects.select(Selectors.config),
+          SearchAdapter.combineNavigations(results)
+        );
+        yield effects.put(<any>flux.actions.receivePastPurchaseRefinements(navigations));
+      }
+    } catch (e) {
+      return effects.put(flux.actions.receivePastPurchaseRefinements(e));
```

Comment:
Should I add this action to the undo error actions? @jmbtutor

Reply:
If it affects the state, then yes, which it looks like it does.

Reply:
just kidding it's already there